exporter/fileexporter/factory.go

```diff
@@ -58,6 +58,7 @@ func createDefaultConfig() config.Exporter {
 	return &Config{
 		ExporterSettings: config.NewExporterSettings(config.NewComponentID(typeStr)),
 		FormatType:       formatTypeJSON,
+		Rotation:         &Rotation{MaxBackups: defaultMaxBackups},
```

Comment:
Do we care that this is not in fact the default configuration? I can appreciate that an unmarshaled config ends up in the correct state, but aren't we losing some clarity by having `createDefaultConfig` return something that needs to be understood as something other than the default configuration?

Reply:
We have that problem on the OTLP receiver https://github.com/open-telemetry/opentelemetry-collector/blob/a9f41a2fbcd713682bc693ce0fba4cd8c948bde7/receiver/otlpreceiver/config.go#L75-L81, the Jaeger receiver https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/d0ae93589ca622d7879b91edb4d4327188ba8497/receiver/jaegerreceiver/config.go#L158-L169 and possibly others. I don't love it either, but it seems to be an established pattern on this codebase.